phits-tech/website
# [People] Data-Driven Heroes & Profiles

## 💯 DoD

- [x] All work is complete
- [x] Migrations: NA
- [x] Release notes: NA
- [x] Issues linked: resolves #7; resolves #8

## 📝 Summary

**General** (not sure why I didn't branch it)
- Ported some _settings_, _lint_, _build_ fixes (from CC)

**Heroes/Profile**
- Type updates:
  - `EventLog` (record of attendance & contribution)
  - `User` (profile, links, contribution)
  - `UserPrivate` (support multiple emails & networks)
  - `NewBase<T>`, `UpdateBase<T>` (support admin FieldValue & FieldValue for optionals)
- `dao.createUser`
- `people` test-data for seeder
- Update Heroes & Profile pages
  - Make both data-driven
  - Pretty colored SVG icons for user network links
  - Pretty shine effect on buttons

**Hint: You have to click to see the diffs for Heroes.html & Profile.html**

## 💡 More ideas

- Redesign profile page (single col on tablet & cleanup responsive grid)
- Multiple roles for one event

## 📸 Screenshots

<img width="840" alt="Heroes" src="https://user-images.githubusercontent.com/31621095/119150733-ec67ce80-ba78-11eb-818a-5551abe75b1f.png">

<img width="840" alt="Profile" src="https://user-images.githubusercontent.com/31621095/119156746-bb8a9800-ba7e-11eb-9a8f-f45abc5ae9a7.png">
